tests/test_constants.py

```diff
@@ -21,39 +37,419 @@ def test_constants(self):
         """
         self.maxDiff = None
 
-        self.assertEqual(constants.CMD_CURRENT_APP, "CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP")
+        # CMD_CURRENT_APP
+        self.assertEqual(constants.CMD_CURRENT_APP, "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\}*} && echo $CURRENT_APP")
```

Comment:
@martomo should there be a `;` or `&&` between these two commands?

```
CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * }
```

Reply:
Interesting catch, technically it seems that variable (re)declarations do not need to be chained (`&&`), as the string manipulation does not affect the exit code or trigger failure.

So it can either be `&&`, `;` or nothing in between variable (re)declarations. Although it might be more consistent (for the time being) to go with `&&`, in case previous executed command did fail when chaining commands, if not it will at least improve readability.

Reply:
Thanks for the explanation. I'll add `&&` tomorrow.